src/Filters/design.jl

```diff
@@ -335,18 +350,20 @@ function transform_prototype(ftype::Highpass, proto::ZeroPoleGain)
 
     abs(real(num) - 1) < np*eps(real(num)) && (num = 1)
     abs(real(den) - 1) < np*eps(real(den)) && (den = 1)
-    ZeroPoleGain(newz, newp, proto.k * real(num)/real(den))
+    ZeroPoleGain(newz, newp, oftype(k, k * real(num)/real(den)))
```

Comment:
It would be the same argument in #142, wouldn't it?
We can call an explicit error saying that this filter cannot be represented by an `Int` gain if that helps.

Reply:
No I mean at this line there is a potential `InexactError`:

```julia
ulia> oftype(1, 1 / 2)
ERROR: InexactError: Int64(0.5)
Stacktrace:
 [1] Type at ./float.jl:703 [inlined]
 [2] convert at ./number.jl:7 [inlined]
 [3] oftype(::Int64, ::Float64) at ./essentials.jl:334
 [4] top-level scope at none:0
```

Reply:
Im sorry, I misspoke 😅.   
I was trying to say that, similar to the case discussed in #142, if the user inputs a filter with `Int` gain, they must have a preference to that. (Assuming we do default construction to `K::Float64`, which is partially the case since the prototype functions generate a gain for the same "real-type" as the poles).   
So the `InexactError` is a consequence of asking for an integer gain when one cannot maintain the same gain as the prototype.

Reply:
IMO preserving types is great, where possible. However, I think it's better to have a function that does not preserve input types, rather than a function that throws errors or intentionally does not work for some input types. Of course, if there is a compelling reason for why it's necessary in this case, then that would be different. What is the problem that you were experiencing or envisioning that motivated this change? I also don't have a good sense for how often people construct zero pole gain filters with integer datatypes, so maybe I'm just wringing my hands over a very niche corner case?

Maybe you could just add a type bound on the filter object to limit these functions to gain types that are closed under division?

Reply:
It was more of a motivation to preserve typing. Similar to how Julia errors for `Int(3.4)`, the assumption I have is the user explicitly creates a filter with the types they want. And if they create a filter with an integer gain, then they must have explicitly wanted one.

But I definitely think that some people want to have integer-only filters

Reply:
TBH I doubt these functions would work with integer data types before this PR. `newz[i] = ftype.w / z[i]` would also be problematic.

Reply:
it might be worthwhile to figure out what people typically expect when they construct a filter and the use cases they have. I am curious if someone would create an integer filter and then modify that, or do all the design work with floats/fixed point and then convert to integers at the end

Reply:
Since it's clear that this function would not have worked with integers in the first place, I'm ok with it still not working with integers, for now.